### DISCUSSION TITLE
Why were accessible, accessibleState and accessibleValue removed?

### DISCUSSION THREAD

**Discussion by finsterwalder** [necolas/react-native-web#2372] [Q&A]
ReactNative has AccessibilityProps accessible, accessibleState and accessibleValue.
They used to be available in react native web 0.17 and have been removed in 0.18.
They have been deprecated and removed now, but I can't find an explanation why.

I found that react native web now has replacements in the form of its own accessibility attributes, like accessibilityChecked: true instead of accessibleState: { checked }.
Doesn't that make react native web incompatible to react native? At least for me developing on both platforms, I now have to put in different accessibility attributes for the different platforms, which used to "just work" before.

And what is a replacement for accessible?

**Comment by necolas** [comment]
https://github.com/necolas/react-native-web/releases

**Reply by finsterwalder** [reply]
Yes, I read throw the release notes. But that does not explain why. Or did I overlook something? then please point me to that directly, please.
And I still could not figure out, whether there is a replacement for the react native "accessible"-element.

**Reply by intergalacticspacehighway** [reply]
There's an [RFC](https://github.com/react-native-community/discussions-and-proposals/pull/496) in react native to support `aria-*` props so we can be aligned with the web spec on native. In the meanwhile, you can use [accessibility*](https://necolas.github.io/react-native-web/docs/accessibility/) mapping API on react native web.

**Reply by penx** [reply]
I just raised https://github.com/necolas/react-native-web/issues/2403 and then found this discussion

My understanding is that

- the [RFC](https://github.com/react-native-community/discussions-and-proposals/pull/496) looks great, but it was opened in August is still in proposal state
- [released versions of react-native](https://reactnative.dev/docs/accessibility) do not have props such as accessibilityChecked
- if I have code that is type checked against react-native and compiled to both native and web, props such as accessibilityChecked are not an option
- the [document linked to above](https://necolas.github.io/react-native-web/docs/accessibility/) states "Additional compatibility with React Native accessibility props is also included"
- accessibilityState was [removed in March](https://github.com/necolas/react-native-web/commit/174ebb3c5d68bd396943a949d5e0098840505c1e) (5 months before the RFC) on the basis that it was deprecated

So how do we use accessibility props such as `checked` in a way that works on both native and web?

**Reply by penx** [reply]
As noted at https://github.com/necolas/react-native-web/issues/2403#issuecomment-1268240837, I've created a library called  [react-native-accessibility-props](https://github.com/penx/react-native-accessibility-props) that can be used like this


```tsx
import { Pressable } from "react-native";
import { getAccessibilityStateProps } from "react-native-accessibility-props";

export default function App() {
  return (
    <Pressable
      accessibilityRole="radio"
      accessibilityLabel="label"
      {...getAccessibilityStateProps({ checked: true })}
    >
      Example
    </Pressable>
  );
}
```

Seems to work, other than the TypeScript types, which I'll think about and hopefully publish an update. PRs welcome if you have an idea!

**Reply by penx** [reply]
...for TypeScript types, I'm able to use [declaration merging as described here](https://github.com/necolas/react-native-web/issues/1684#issuecomment-712297248), e.g.

```ts
// types/react-native.d.ts
import "react-native";

declare module "react-native" {
  /**
   * react-native with react-native-web extensions
   * @see https://necolas.github.io/react-native-web/docs/accessibility/#accessibility-props-api
   */
  export interface AccessibilityProps {
    /**
     * Equivalent to [aria-checked](https://www.w3.org/TR/wai-aria-1.2/#aria-checked).
     */
    accessibilityChecked?: ?boolean;
  }
}

```
